### ISSUE TITLE
unnecessary call to min function

### ISSUE THREAD

**Issue by FelixMaetzler** [rust-lang/rust-clippy#11924]
### What it does

A few days ago i opened #11901 
You could do this for the `min()` method too.
So `0.min(<anything>)` is always zero if `<anything>` is an unsigned integer.
Also this method is symmetric, that means same applies to `<anything>.min(0)`.

You could go even further and generalize this to `T::MIN.min(<anything>)` (where `T` is any type of integer like i.e. `i64`) is always `T::MIN`.
And you could do this with the upper  bound too:  `T::MAX.min(<anything>)` is for any integer type `T` the same as `<anything>`.

### Advantage

- Remove of unnecessary function call
- Reduction of Code Complexity

### Example

```rust
// example with zero and unsigned Type
let a = 0.min(42_usize);

// example with MIN and signed Type
let a = 42.min(i64::MIN);

// example with MAX and signed Type
let a = i32::MAX.min(9);
```

Could be written as:

```rust
// example with zero and unsigned Type
let a = 0;

// example with MIN and signed Type
let a = i64::MIN;

// example with MAX and signed Type
let a = 9;
```

**Comment by Centri3** [comment]
> I think i don't quite understand what you're saying.

It's 4am. On the first section I mean that, rather than only linting when the receiver is `T::MIN`, we always lint it if they're both constants and `recv < arg` and they're both constants (or literals).

The second section is how `constant` turns MIR (external)/HIR (same crate) into, well, a `Constant`. It doesn't look into any (many?) function calls, so for example, `0u32.min(a())` wouldn't lint since it wouldn't evaluate to a constant. This isn't good, so if it's `T::MIN` (and `T::MAX` for `max`), the lint shouldn't check the argument at all.

**Comment by Centri3** [comment]
> My bad.

Was just explaining why it was worded improperly 😅 You're fine :)

**Comment by FelixMaetzler** [comment]
So i have implemented the first draft. Because i'm new to colloboration on github, what is the next step?
How can i `show` my code so that other people can have a look at it and give feedback and check it for bugs etc?

**Comment by y21** [comment]
Closing as this was implemented in #12368
